federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -176,25 +279,42 @@ func (ic *IngressController) Run(stopChan <-chan struct{}) {
 	go ic.ingressInformerController.Run(stopChan)
 	glog.Infof("... Starting Ingress Federated Informer")
 	ic.ingressFederatedInformer.Start()
+	glog.Infof("... Starting ConfigMap Federated Informer")
+	ic.configMapFederatedInformer.Start()
 	go func() {
 		<-stopChan
-		glog.Infof("Stopping Ingress Controller")
+		glog.Infof("Stopping Ingress Federated Informer")
 		ic.ingressFederatedInformer.Stop()
+		glog.Infof("Stopping ConfigMap Federated Informer")
+		ic.configMapFederatedInformer.Stop()
 	}()
 	ic.ingressDeliverer.StartWithHandler(func(item *util.DelayingDelivererItem) {
 		ingress := item.Value.(types.NamespacedName)
 		glog.V(4).Infof("Ingress change delivered, reconciling: %v", ingress)
 		ic.reconcileIngress(ingress)
 	})
-	ic.clusterDeliverer.StartWithHandler(func(_ *util.DelayingDelivererItem) {
-		glog.V(4).Infof("Cluster change delivered, reconciling ingresses")
-		ic.reconcileIngressesOnClusterChange()
+	ic.clusterDeliverer.StartWithHandler(func(item *util.DelayingDelivererItem) {
+		clusterName := item.Key
+		if clusterName != allClustersKey {
+			glog.V(4).Infof("Cluster change delivered for cluster %q, reconciling configmap and ingress for that cluster", clusterName)
+		} else {
+			glog.V(4).Infof("Cluster change delivered for all clusters, reconciling configmaps and ingresses for all clusters")
+		}
+		ic.reconcileIngressesOnClusterChange(clusterName)
+		ic.reconcileConfigMapForCluster(clusterName)
+	})
+	ic.configMapDeliverer.StartWithHandler(func(item *util.DelayingDelivererItem) {
+		clusterName := item.Key
+		glog.V(4).Infof("ConfigMap change delivered for cluster %q, reconciling configmap for that cluster", clusterName)
```

Comment:
No, this is somewhat by design.  ReplicaSets and Ingresses need to be reconciled across all clusters when the cluster set changes, while for ConfigMaps it's a simple matter of reconciling for the one changed cluster.